metadata_service/proxy/neo4j_proxy.py

```diff
@@ -173,7 +174,8 @@ def _exec_table_query(self, table_uri: str) -> Tuple:
         """)
 
         table_records = self._execute_cypher_query(statement=table_level_query,
-                                                   param_dict={'tbl_key': table_uri})
+                                                   param_dict={'tbl_key': table_uri,
+                                                               'tag_type': 'default'})
```

Comment:
I would like to avoid making a separate call to fetch badges for each table when they could be included in the base result just like tags. This problem gets worse with search results where we'd have to make 10 queries at once (one for each search result).

Not sure about the exact syntax but could we do something like
```
  ...
  OPTIONAL MATCH (tbl)-[:TAGGED_BY]->(tag:Tag)
  ...
  collect(distinct tag:Tag{tag_type:'default'}) as tag_records,
  collect(distinct tag:Tag{tag_type:'badge'}) as badge_records,
```
We would also remove the `tag_type` parameter from `param_dict`.

Reply:
Or would the syntax be 

```
...
  OPTIONAL MATCH (tbl)-[:TAGGED_BY]->(tag:Tag{tag_type: 'default'})
  OPTIONAL MATCH (tbl)-[:TAGGED_BY]->(badge:Tag{tag_type: 'badge'})
  ...
  collect(distinct tag) as tag_records,
  collect(distinct badge) as badge_records,

```

Reply:
I think putting everything inside single call won't fly. What if we later need to add table_stats to show the table object?

Reply:
but if you have strong opinion this one, I could modify on Friday or early next week. WDYT @jinhyukchang

Reply:
I don't see a strong reason why `badges` should be different from `tags` or why `table_stats` should be different from `col_stats`.

Calling `get_table` currently queries for tags, owners, columns, and multiple `col_stats` for each column, which can result in hundreds of `col_stat` objects. If we're concerned about adding `table_stats` or `badges`, the performance cost will be a small fraction of whatever `col_stats` costs. If we were to draw an arbitrary line about what data gets included in the base API and what gets left off, I'd argue that `col_stats` gets left off and everything else gets added in. Currently fetching table metadata takes between 100-200 ms so I don't think this is a concern yet.

On the flip side let's say we want to decomp all of these APIs. On loading the `table_details` page we would make calls to `get_table`, `get_table_tags`, `get_table_badges`, `get_table_stats`, `get_table_columns` and `get_column_stats`. I don't think this kind of micro-optimization is worth the effort.

Reply:
Can we try query like below?:
...
OPTIONAL MATCH (tbl)-[:TAGGED_BY]->(tag:Tag{tag_type: 'default'})
OPTIONAL MATCH (tbl)-[:TAGGED_BY]->(badge:Tag{tag_type: 'badge'})
...

Reply:
Oops. It's already mentioned. Shall we see how it goes? I believe we still have sometime, right?